### DISCUSSION TITLE
Porting Xamarin.Form Control to Uno.*

### DISCUSSION THREAD

**Discussion by workgroupengineering** [unoplatform/uno#6541] [Q&A]
Hi,
Can you give me suggestions on how to migrate a control created for Xamarin.Form into Uno.*?

**Comment by robloo** [comment]
I think we need more details on how your custom control code functions. If you are rendering a chart in Skia for example, a port will be relatively straightforward. If you are doing a lot of custom drawings using Xamarin.Form techniques it's basically a full re-write. Uno Platform uses the WinUI control architecture and the two are fundamentally quite different as I'm sure you know.

That said, it may be possible to actually embed a Xamarin.Forms control in an Uno Platform app (similar to using Xamarin Native controls). Someone with a lot more knowledge than me will need to comment further on this though.

**Reply by workgroupengineering** [reply]
Thanks for the reply, I'm trying to migrate XCT's CameraView to Uno.WinUI.
The main need is to understand how to make the native controls interact with Uno, especially to understand how the arrangement takes place.
